### ISSUE TITLE
Routes still appear in browser back button history after setting `keepHistory: false` in a web app

### ISSUE THREAD

**Issue by tbuczkowski** [Milad-Akarie/auto_route_library#1678]
Hi, I'm trying to solve a similar issue to the one described in https://github.com/Milad-Akarie/auto_route_library/issues/1378. 

Trying to prevent the user being able to navigate using the browser back button, I have set `keepHistory: false` on all my routes. This does not seem to work correctly - while navigating through the app, the back button history keeps getting updated with each route. This happens both when navigating through `.push()` and `.replaceAll()`.

After some debugging, I have noticed, that the only place where the `keepHistory` parameter is being accessed in the whole package is the `_addEntry` method in the `routing_controller.dart` file: 

```
  Future<T?> _addEntry<T extends Object?>(
    RouteMatch route, {
    bool notify = true,
  }) {
    final topRoute = _pages.lastOrNull?.routeData;
    if (topRoute != null && topRoute._match.keepHistory == false) {
      markUrlStateForReplace();
      _removeRoute(topRoute._match, notify: false);
    }
    final data = _createRouteData(route, routeData);
    final page = pageBuilder(data);
    _pages.add(page);
    if (notify) {
      notifyAll();
    }
    return (page as AutoRoutePage<T>).popped;
  }
  ```  

I discovered that inside that method, the `topRoute` is always null in my case. That causes it to never call the `_removeRoute` method, which I presume would then remove the page from navigation history.

The versions I'm using:
Flutter: 3.10.6
auto_route: 7.8.0
auto_route_generator: 7.3.0

**Comment by Milad-Akarie** [comment]
Hey @tbuczkowski what's the visual status for when topRoute is null? do you have zero routes in stack?

**Comment by tbuczkowski** [comment]
No, I'm navigating normally through the app, and the `topRoute` always is null, no matter how/where I navigate.

**Comment by B0yma** [comment]
i have the same problem with vers auto_route: 7.8.4
